## Este notebook genera la limpieza de las fuentes de It cloud del CRM, eventos y Facturacion desde el año 2013-2020 y genera un archivo consolidado de las 3 fuentes.

###### Este notebook fue creado por Francy Pineda estudiante de Maestria MINE 2018

###### Se importan librerias necesarias

In [161]:
import pandas as pd
import numpy as np
import matplotlib
from matplotlib import pyplot as plt
%matplotlib inline
matplotlib.style.use('ggplot')
import seaborn as sns
import os
from dateutil.parser import parse
pd.options.display.max_rows = 350
pd.options.display.max_columns = 350
from scipy import stats


##### Fuentes crm_clientes y eventos

In [162]:
#%cd /home/xubuntu/Documents/ProyectoFinalMINE/Datos/crm_eventos
%cd D:\MINE\ProyectoFinalMINE\Datos\crm_eventos
crm_eventos = pd.read_excel('eventos_del_crm.xlsx')

D:\MINE\ProyectoFinalMINE\Datos\crm_eventos


In [163]:
#Se usa la salida de campos estandarizados del CRM completo, ciudad y prioridad
#%cd /home/xubuntu/Documents/ProyectoFinalMINE/Datos/Salidas
%cd D:\MINE\ProyectoFinalMINE\Datos\Fuentes Usadas Tableau\salidas
crm_clientes = pd.read_excel('crm_clientes_sector_v2.xlsx')

D:\MINE\ProyectoFinalMINE\Datos\Fuentes Usadas Tableau\salidas


##### Fuentes Facturacion 2013

In [164]:
#%cd   /home/xubuntu/Documents/ProyectoFinalMINE/Datos/facturacion/2013
% cd D:\MINE\ProyectoFinalMINE\Datos\facturacion\2013
facturacion_julio_2013 = pd.read_excel('Julio2013.xls')
facturacion_agosto_2013 = pd.read_excel('Agosto2013.xls')
facturacion_septiembre_2013 = pd.read_excel('Septiembre2013.xls')
facturacion_octubre_2013 = pd.read_excel('Octubre2013.xls')
facturacion_noviembre_2013 = pd.read_excel('Novimbre2013.xls')
facturacion_diciembre_2013 = pd.read_excel('Diciembre2013.xls')

D:\MINE\ProyectoFinalMINE\Datos\facturacion\2013


##### Fuentes Facturacion 2014

In [165]:
#%cd   /home/xubuntu/Documents/ProyectoFinalMINE/Datos/facturacion/2014
% cd D:\MINE\ProyectoFinalMINE\Datos\facturacion\2014
facturacion_enero_2014 = pd.read_excel('Enero 2014.xls')
facturacion_febrero_2014 = pd.read_excel('febrero 2014.xls')
facturacion_marzo_2014 = pd.read_excel('Marzo2014.xls')
facturacion_abril_2014 = pd.read_excel('Abril2014.xls')
facturacion_mayo_2014 = pd.read_excel('mayo2014.xls')
facturacion_junio_2014 = pd.read_excel('junio2014.xls')
facturacion_julio_2014 = pd.read_excel('julio2014.xls')
facturacion_agosto_2014 = pd.read_excel('Agosto2014.xls')
facturacion_septiembre_2014 = pd.read_excel('Septiembre2014.xls')
facturacion_octubre_2014 = pd.read_excel('Octubre2014.xls')
facturacion_noviembre_2014 = pd.read_excel('noviembre2014.xls')
facturacion_diciembre_2014 = pd.read_excel('Diciembre2014.xls')

D:\MINE\ProyectoFinalMINE\Datos\facturacion\2014


##### Fuentes Facturacion 2015

In [166]:
#%cd /home/xubuntu/Documents/ProyectoFinalMINE/Datos/facturacion/2015
% cd D:\MINE\ProyectoFinalMINE\Datos\facturacion\2015
facturacion_enero_2015 = pd.read_excel('Enero2015.xls')
facturacion_febrero_2015 = pd.read_excel('Febrero2015.xls')
facturacion_marzo_2015 = pd.read_excel('Marzo2015.xls')
facturacion_abril_2015 = pd.read_excel('Abril2015.xls')
facturacion_mayo_2015 = pd.read_excel('Mayo2015.xls')
facturacion_junio_2015 = pd.read_excel('Junio2015.xls')
facturacion_julio_2015 = pd.read_excel('Julio2015.xls')
facturacion_agosto_2015 = pd.read_excel('Agosto2015.xls')
facturacion_septiembre_2015 = pd.read_excel('Septiembre2015.xls')
facturacion_octubre_2015 = pd.read_excel('Octubre2015.xls')
facturacion_noviembre_2015 = pd.read_excel('Noviembre2015.xls')
facturacion_diciembre_2015 = pd.read_excel('Diciembre2015.xls')

D:\MINE\ProyectoFinalMINE\Datos\facturacion\2015


##### Fuentes Facturacion 2016

In [167]:
#%cd   /home/xubuntu/Documents/ProyectoFinalMINE/Datos/facturacion/2016
% cd D:\MINE\ProyectoFinalMINE\Datos\facturacion\2016
facturacion_enero_2016 = pd.read_excel('Enero2016.xls')
facturacion_febrero_2016 = pd.read_excel('Febrero2016.xls')
facturacion_marzo_2016 = pd.read_excel('MArzo2016.xls')
facturacion_abril_2016 = pd.read_excel('Abril2016.xls')
facturacion_mayo_2016 = pd.read_excel('MAyo2016.xls')
facturacion_junio_2016 = pd.read_excel('junio2016.xls')
facturacion_julio_2016 = pd.read_excel('Julio2016.xls')
facturacion_agosto_2016 = pd.read_excel('Agosto2016.xls')
facturacion_septiembre_2016 = pd.read_excel('Septiembre2016.xls')
facturacion_octubre_2016 = pd.read_excel('Octubre2016.xls')
facturacion_noviembre_2016 = pd.read_excel('Noviembre2016.xls')
facturacion_diciembre_2016 = pd.read_excel('Diciembre2016.xls')

D:\MINE\ProyectoFinalMINE\Datos\facturacion\2016


##### Fuentes Facturacion 2017

In [168]:
#%cd    /home/xubuntu/Documents/ProyectoFinalMINE/Datos/facturacion/2017
% cd D:\MINE\ProyectoFinalMINE\Datos\facturacion\2017
facturacion_enero_2017 = pd.read_excel('Enero2017.xls')
facturacion_febrero_2017 = pd.read_excel('Febrero2017.xls')
facturacion_marzo_2017 = pd.read_excel('Marzo2017.xls')
facturacion_abril_2017 = pd.read_excel('Abril2017.xls')
facturacion_mayo_2017 = pd.read_excel('Mayo2017.xls')
facturacion_junio_2017 = pd.read_excel('Junio2017.xls')
facturacion_julio_2017 = pd.read_excel('Julio2017.xls')
facturacion_agosto_2017 = pd.read_excel('Agosto2017.xls')
facturacion_septiembre_2017 = pd.read_excel('Septiembre2017.xls')
facturacion_octubre_2017 = pd.read_excel('Octubre2017.xls')
facturacion_noviembre_2017 = pd.read_excel('Noviembre2017.xls')
facturacion_diciembre_2017 = pd.read_excel('Diciembre2017.xls')

D:\MINE\ProyectoFinalMINE\Datos\facturacion\2017


##### Fuentes Facturacion 2018

In [169]:
#%cd /home/xubuntu/Documents/ProyectoFinalMINE/Datos/facturacion/2018
% cd D:\MINE\ProyectoFinalMINE\Datos\facturacion\2018
facturacion_enero_2018 = pd.read_excel('Enero2018.xls')
facturacion_febrero_2018 = pd.read_excel('Febrero2018.xls')
facturacion_marzo_2018 = pd.read_excel('Marzo2018.xls')
facturacion_abril_2018 = pd.read_excel('Abril2018.xls')
facturacion_mayo_2018 = pd.read_excel('Mayo2018.xls')
facturacion_junio_2018 = pd.read_excel('Junio2018.xls')
facturacion_julio_2018 = pd.read_excel('Julio2018.xls')
facturacion_agosto_2018 = pd.read_excel('Agosto2018.xls')
facturacion_septiembre_2018 = pd.read_excel('Septiembre2018.xls')
facturacion_octubre_2018 = pd.read_excel('Octubre2018.xls')
facturacion_noviembre_2018 = pd.read_excel('Noviembre2018.xls')
facturacion_diciembre_2018 = pd.read_excel('Diciembre2018.xls')

D:\MINE\ProyectoFinalMINE\Datos\facturacion\2018


##### Fuentes Facturacion 2019

In [170]:
#%cd     /home/xubuntu/Documents/ProyectoFinalMINE/Datos/facturacion/2019
% cd D:\MINE\ProyectoFinalMINE\Datos\facturacion\2019
facturacion_enero_2019 = pd.read_excel('Enero2019.xls')
facturacion_febrero_2019 = pd.read_excel('Febrero2019.xls')
facturacion_marzo_2019 = pd.read_excel('Marzo2019.xls')
facturacion_abril_2019 = pd.read_excel('Abril2019.xls')
facturacion_mayo_2019 = pd.read_excel('Mayo2019.xls')
facturacion_junio_2019 = pd.read_excel('Junio2019.xls')
facturacion_julio_2019 = pd.read_excel('Julio2019.xls')
facturacion_agosto_2019 = pd.read_excel('Agosto2019.xls')
facturacion_septiembre_2019 = pd.read_excel('Septiembre2019.xls')
facturacion_octubre_2019 = pd.read_excel('Octubre2019.xls')
facturacion_noviembre_2019 = pd.read_excel('Noviembre2019.xls')
facturacion_diciembre_2019 = pd.read_excel('Diciembre2019.xls')

D:\MINE\ProyectoFinalMINE\Datos\facturacion\2019


##### Fuentes Facturacion 2020

In [171]:
#%cd     /home/xubuntu/Documents/ProyectoFinalMINE/Datos/facturacion/2020
% cd D:\MINE\ProyectoFinalMINE\Datos\facturacion\2020
facturacion_enero_2020 = pd.read_excel('Enero2020.xls')
facturacion_febrero_2020 = pd.read_excel('Febrero2020.xls')
facturacion_marzo_2020 = pd.read_excel('Marzo2020.xls')
facturacion_abril_2020 = pd.read_excel('Abril2020.xls')
facturacion_mayo_2020 = pd.read_excel('Mayo2020.xls')
facturacion_junio_2020 = pd.read_excel('Junio2020.xls')
facturacion_julio_2020 = pd.read_excel('Julio2020.xls')
facturacion_agosto_2020 = pd.read_excel('Agosto2020.xls')
facturacion_septiembre_2020 = pd.read_excel('Septiembre2020.xls')
facturacion_octubre_2020 = pd.read_excel('Octubre2020.xls')
facturacion_noviembre_2020 = pd.read_excel('Noviembre2020.xls')
facturacion_diciembre_2020 = pd.read_excel('Diciembre2020.xls')

D:\MINE\ProyectoFinalMINE\Datos\facturacion\2020


##### CONCATENANDO LAS FUENTES DE FACTURACION POR AÑO

In [172]:
frames20 = [facturacion_enero_2020,facturacion_febrero_2020,facturacion_marzo_2020,facturacion_abril_2020 ,facturacion_mayo_2020 ,facturacion_junio_2020,facturacion_julio_2020,facturacion_agosto_2020,facturacion_septiembre_2020,facturacion_octubre_2020,facturacion_noviembre_2020,facturacion_diciembre_2020 ] 
frames19 = [facturacion_enero_2019,facturacion_febrero_2019,facturacion_marzo_2019,facturacion_abril_2019 ,facturacion_mayo_2019 ,facturacion_junio_2019,facturacion_julio_2019,facturacion_agosto_2019,facturacion_septiembre_2019,facturacion_octubre_2019,facturacion_noviembre_2019,facturacion_diciembre_2019 ]
frames18 = [facturacion_enero_2018,facturacion_febrero_2018,facturacion_marzo_2018,facturacion_abril_2018 ,facturacion_mayo_2018 ,facturacion_junio_2018,facturacion_julio_2018,facturacion_agosto_2018,facturacion_septiembre_2018,facturacion_octubre_2018,facturacion_noviembre_2018,facturacion_diciembre_2018 ]
frames17 = [facturacion_enero_2017,facturacion_febrero_2017,facturacion_marzo_2017,facturacion_abril_2017 ,facturacion_mayo_2017 ,facturacion_junio_2017,facturacion_julio_2017,facturacion_agosto_2017,facturacion_septiembre_2017,facturacion_octubre_2017,facturacion_noviembre_2017,facturacion_diciembre_2017 ] 
frames16 = [facturacion_enero_2016,facturacion_febrero_2016,facturacion_marzo_2016,facturacion_abril_2016 ,facturacion_mayo_2016 ,facturacion_junio_2016,facturacion_julio_2016,facturacion_agosto_2016,facturacion_septiembre_2016,facturacion_octubre_2016,facturacion_noviembre_2016,facturacion_diciembre_2016 ]
frames15 = [facturacion_enero_2015,facturacion_febrero_2015,facturacion_marzo_2015,facturacion_abril_2015 ,facturacion_mayo_2015 ,facturacion_junio_2015,facturacion_julio_2015,facturacion_agosto_2015,facturacion_septiembre_2015,facturacion_octubre_2015,facturacion_noviembre_2015,facturacion_diciembre_2015 ]
frames14 = [facturacion_enero_2014,facturacion_febrero_2014,facturacion_marzo_2014,facturacion_abril_2014 ,facturacion_mayo_2014 ,facturacion_junio_2014,facturacion_julio_2014,facturacion_agosto_2014,facturacion_septiembre_2014,facturacion_octubre_2014,facturacion_noviembre_2014,facturacion_diciembre_2014 ]
frames13 = [facturacion_julio_2013,facturacion_agosto_2013,facturacion_septiembre_2013,facturacion_octubre_2013,facturacion_noviembre_2013,facturacion_diciembre_2013 ]

fact_2020 = pd.concat(frames20)
fact_2019 = pd.concat(frames19)
fact_2018 = pd.concat(frames18)
fact_2017 = pd.concat(frames17)
fact_2016 = pd.concat(frames16)
fact_2015 = pd.concat(frames15)
fact_2014 = pd.concat(frames14)
fact_2013 = pd.concat(frames13)
frames=[fact_2013,fact_2014,fact_2015,fact_2016,fact_2017,fact_2018,fact_2019,fact_2020]
facturacion_total = pd.concat(frames) # Contiene toda la informacion de facturacion

In [175]:
facturacion_total["Correos Electrónicos Cliente"].nunique()

452

In [110]:
#%cd     /home/xubuntu/Documents/ProyectoFinalMINE/Datos/Salidas
%cd D:\MINE\ProyectoFinalMINE\Datos\Fuentes Usadas Tableau\salidas
facturacion_total.to_excel("facturacion_completa.xlsx")

D:\MINE\ProyectoFinalMINE\Datos\Fuentes Usadas Tableau\salidas


In [111]:
facturacion_total.head(2)

Tipo Doc. Pre  No. Doc.  Orden/Lugar Entrega  ID Empresa Tipo de Factura  \
0   Factura  IT       3.0                  NaN        98.0     Electrónica   
1   Factura  IT       2.0                  NaN        98.0     Electrónica   

      Estado  ID. Cliente ID. Int. Cliente   ID Interno  \
0    Anulada  9.00227e+08      9.00227e+08  9.00227e+08   
1  Entregada  9.00227e+08      9.00227e+08  9.00227e+08   

               Nombre del Cliente     Año  Mes Fecha Exp.  \
0  Contact Center Colombia C3 SAS  2013.0  7.0 2013-07-22   
1  Contact Center Colombia C3 SAS  2013.0  7.0 2013-07-19   

    Fecha Hora Generación Moneda        Vendedor  Cód. Vendedor    SubTotal  \
0 2013-07-22 19:38:35.820    COP      Indefinido            NaN    900000.0   
1 2013-07-19 21:11:20.777    COP  Wilson Rojas P            NaN  11353185.0   

         IVA  Descuento  Otros  Total Retención       Total  Total A Pagar  \
0   144000.0        0.0    0.0              0.0   1044000.0            NaN   
1  1816509.6        0.0    0.0              0.0  13169694.6            NaN   

        Ver     Ver.1 Estado Notificación     Teléfono  \
0  322926.0  322926.0                 NaN  4.86202e+06   
1  322065.0  322065.0                 NaN  4.86202e+06   

  Correos Electrónicos Cliente Correos Electrónicos Enviados  
0       aduenas@c3colombia.com                           NaN  
1       aduenas@c3colombia.com                           NaN

In [112]:
facturacion_total.columns

Index(['Tipo Doc.', 'Pre', 'No. Doc.', 'Orden/Lugar Entrega', 'ID Empresa',
       'Tipo de Factura', 'Estado', 'ID. Cliente', 'ID. Int. Cliente',
       'ID Interno', 'Nombre del Cliente', 'Año', 'Mes', 'Fecha Exp.',
       'Fecha Hora Generación', 'Moneda', 'Vendedor', 'Cód. Vendedor',
       'SubTotal', 'IVA', 'Descuento', 'Otros', 'Total Retención', 'Total',
       'Total A Pagar', 'Ver', 'Ver.1', 'Estado Notificación', 'Teléfono',
       'Correos Electrónicos Cliente', 'Correos Electrónicos Enviados'],
      dtype='object')

## 1. Quitar los espacios adelante y atrás del atributo CORREOS ELECTRONICOS CLIENTES

In [113]:
facturacion_total['Correos Electrónicos Cliente']  =facturacion_total['Correos Electrónicos Cliente'].str.strip()

## 2. Generar el atributo EMAIL simplemente pasando a mayúsculas el atributo Correos Electrónicos Cliente

In [114]:
facturacion_total['EMAIL'] =facturacion_total['Correos Electrónicos Cliente'].str.upper()

In [115]:
facturacion_total.head(2)

Tipo Doc. Pre  No. Doc.  Orden/Lugar Entrega  ID Empresa Tipo de Factura  \
0   Factura  IT       3.0                  NaN        98.0     Electrónica   
1   Factura  IT       2.0                  NaN        98.0     Electrónica   

      Estado  ID. Cliente ID. Int. Cliente   ID Interno  \
0    Anulada  9.00227e+08      9.00227e+08  9.00227e+08   
1  Entregada  9.00227e+08      9.00227e+08  9.00227e+08   

               Nombre del Cliente     Año  Mes Fecha Exp.  \
0  Contact Center Colombia C3 SAS  2013.0  7.0 2013-07-22   
1  Contact Center Colombia C3 SAS  2013.0  7.0 2013-07-19   

    Fecha Hora Generación Moneda        Vendedor  Cód. Vendedor    SubTotal  \
0 2013-07-22 19:38:35.820    COP      Indefinido            NaN    900000.0   
1 2013-07-19 21:11:20.777    COP  Wilson Rojas P            NaN  11353185.0   

         IVA  Descuento  Otros  Total Retención       Total  Total A Pagar  \
0   144000.0        0.0    0.0              0.0   1044000.0            NaN   
1  1816509.6        0.0    0.0              0.0  13169694.6            NaN   

        Ver     Ver.1 Estado Notificación     Teléfono  \
0  322926.0  322926.0                 NaN  4.86202e+06   
1  322065.0  322065.0                 NaN  4.86202e+06   

  Correos Electrónicos Cliente Correos Electrónicos Enviados  \
0       aduenas@c3colombia.com                           NaN   
1       aduenas@c3colombia.com                           NaN   

                    EMAIL  
0  ADUENAS@C3COLOMBIA.COM  
1  ADUENAS@C3COLOMBIA.COM

In [116]:
facturacion_total_contodos_atributos=facturacion_total

In [117]:
facturacion_total_contodos_atributos.head(2)

Tipo Doc. Pre  No. Doc.  Orden/Lugar Entrega  ID Empresa Tipo de Factura  \
0   Factura  IT       3.0                  NaN        98.0     Electrónica   
1   Factura  IT       2.0                  NaN        98.0     Electrónica   

      Estado  ID. Cliente ID. Int. Cliente   ID Interno  \
0    Anulada  9.00227e+08      9.00227e+08  9.00227e+08   
1  Entregada  9.00227e+08      9.00227e+08  9.00227e+08   

               Nombre del Cliente     Año  Mes Fecha Exp.  \
0  Contact Center Colombia C3 SAS  2013.0  7.0 2013-07-22   
1  Contact Center Colombia C3 SAS  2013.0  7.0 2013-07-19   

    Fecha Hora Generación Moneda        Vendedor  Cód. Vendedor    SubTotal  \
0 2013-07-22 19:38:35.820    COP      Indefinido            NaN    900000.0   
1 2013-07-19 21:11:20.777    COP  Wilson Rojas P            NaN  11353185.0   

         IVA  Descuento  Otros  Total Retención       Total  Total A Pagar  \
0   144000.0        0.0    0.0              0.0   1044000.0            NaN   
1  1816509.6        0.0    0.0              0.0  13169694.6            NaN   

        Ver     Ver.1 Estado Notificación     Teléfono  \
0  322926.0  322926.0                 NaN  4.86202e+06   
1  322065.0  322065.0                 NaN  4.86202e+06   

  Correos Electrónicos Cliente Correos Electrónicos Enviados  \
0       aduenas@c3colombia.com                           NaN   
1       aduenas@c3colombia.com                           NaN   

                    EMAIL  
0  ADUENAS@C3COLOMBIA.COM  
1  ADUENAS@C3COLOMBIA.COM

## 3.Quitar atributos de Facturacion

In [118]:
del facturacion_total['Tipo Doc.']
del facturacion_total['Pre']
del facturacion_total['ID Empresa']
del facturacion_total['Moneda']
del facturacion_total['Otros']
del facturacion_total['Descuento']
del facturacion_total['Vendedor']
del facturacion_total['Ver']
del facturacion_total['Ver.1']
del facturacion_total['ID. Int. Cliente']
del facturacion_total['ID Interno']
del facturacion_total['Nombre del Cliente']
del facturacion_total['No. Doc.']
del facturacion_total['Teléfono']
del facturacion_total['Orden/Lugar Entrega']
del facturacion_total['Cód. Vendedor']
del facturacion_total['Total']
del facturacion_total['Total A Pagar']
del facturacion_total['IVA']
del facturacion_total['Total Retención']
del facturacion_total['Correos Electrónicos Enviados']
del facturacion_total['Correos Electrónicos Cliente']
del facturacion_total['ID. Cliente']

In [119]:
facturacion_total.head(2)

Tipo de Factura     Estado     Año  Mes Fecha Exp.   Fecha Hora Generación  \
0     Electrónica    Anulada  2013.0  7.0 2013-07-22 2013-07-22 19:38:35.820   
1     Electrónica  Entregada  2013.0  7.0 2013-07-19 2013-07-19 21:11:20.777   

     SubTotal Estado Notificación                   EMAIL  
0    900000.0                 NaN  ADUENAS@C3COLOMBIA.COM  
1  11353185.0                 NaN  ADUENAS@C3COLOMBIA.COM

## Se filtran solo las facturas validas

In [120]:
#facturacion_total=facturacion_total.loc[(facturacion_total['Moneda'] =='COP')]
facturacion_total=facturacion_total.loc[facturacion_total['Estado'].isin(['Entregada', 'Aceptada', 'Creada', 'Modificada'])]

In [121]:
facturacion_por_cliente2= facturacion_total.groupby('EMAIL')['Estado'].count().reset_index()

In [122]:
facturacion_por_cliente2.rename(columns={'Estado': 'Numero_facturas'}, inplace=True)

## BASE COMBINADA

## 1.A Se calcula número de facturas registradas por email de la base de facturación

In [123]:
facturacion_por_cliente2.head(2)

EMAIL  Numero_facturas
0  ABOGADOANDRESLOPEZ@GMAIL.COM               14
1         AC.COMPRAS1@GMAIL.COM                1

## 1.B Se calcula Subtotal acumulado por email de la base de facturación

In [124]:
facturacion_por_cliente3= facturacion_total.groupby('EMAIL')['SubTotal'].sum().reset_index()

In [125]:
facturacion_por_cliente3.head(2)

EMAIL   SubTotal
0  ABOGADOANDRESLOPEZ@GMAIL.COM  1905653.4
1         AC.COMPRAS1@GMAIL.COM   408000.0

In [157]:
facturacion_por_cliente3.shape

(448, 2)

## Se une el numero de facturas y el subtotal para cada cliente

In [126]:
facturacion_num_facturas =  pd.merge(facturacion_por_cliente3, facturacion_por_cliente2, how='inner', left_on=['EMAIL'],right_on=['EMAIL'])
facturacion_num_facturas.head(2)

EMAIL   SubTotal  Numero_facturas
0  ABOGADOANDRESLOPEZ@GMAIL.COM  1905653.4               14
1         AC.COMPRAS1@GMAIL.COM   408000.0                1

In [127]:
facturacion_num_facturas.shape

(448, 3)

## 1.C Se calcula la ULTIMA FECHA de factura POR EMAIL

In [128]:
facturacion_total.head(2)

Tipo de Factura     Estado     Año  Mes Fecha Exp.   Fecha Hora Generación  \
1     Electrónica  Entregada  2013.0  7.0 2013-07-19 2013-07-19 21:11:20.777   
2  Por Computador  Entregada  2013.0  7.0 2013-07-16 2013-07-16 08:18:06.617   

     SubTotal Estado Notificación                   EMAIL  
1  11353185.0                 NaN  ADUENAS@C3COLOMBIA.COM  
2   3934750.0                 NaN  ADUENAS@C3COLOMBIA.COM

In [129]:
facturacion_total["Fecha Exp."] = pd.to_datetime(facturacion_total["Fecha Exp."])

C:\Users\franc\Anaconda3\lib\site-packages\ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [130]:
ultima_facturacion=facturacion_total[["Fecha Exp.","EMAIL"]]

In [131]:
ultima_facturacion=ultima_facturacion.drop_duplicates(subset="EMAIL",keep='last')

In [132]:
ultima_facturacion.rename(columns={'Fecha Exp.': 'ultima_factura'}, inplace=True)

In [133]:
ultima_facturacion.shape

(448, 2)

In [134]:
ultima_facturacion.tail(2)

ultima_factura                                    EMAIL
116     2017-12-01  ADMINISTRATIVO@CATALOGOPUNTOTALL.COM.CO
117     2017-12-01               ANDRES.MORENO@CARVAJAL.COM

In [135]:
facturacion_total.tail(2)

Tipo de Factura      Estado     Año   Mes Fecha Exp.  \
116  Por Computador      Creada  2017.0  12.0 2017-12-01   
117  Por Computador  Modificada  2017.0  12.0 2017-12-01   

      Fecha Hora Generación  SubTotal Estado Notificación  \
116 2017-12-01 08:55:22.470  199998.0                 NaN   
117 2017-12-01 08:43:05.010   98020.0                 NaN   

                                       EMAIL  
116  ADMINISTRATIVO@CATALOGOPUNTOTALL.COM.CO  
117               ANDRES.MORENO@CARVAJAL.COM

In [136]:
ultima_facturacion["EMAIL"].nunique()

448

In [137]:
client_fact_ultima_factura =  pd.merge(facturacion_num_facturas, ultima_facturacion, how='inner', left_on=['EMAIL'],right_on=['EMAIL'])
client_fact_ultima_factura.head(2)

EMAIL   SubTotal  Numero_facturas ultima_factura
0  ABOGADOANDRESLOPEZ@GMAIL.COM  1905653.4               14     2017-12-10
1         AC.COMPRAS1@GMAIL.COM   408000.0                1     2014-12-24

In [138]:
facturacion_num_facturas.shape

(448, 3)

In [139]:
ultima_facturacion.shape

(448, 2)

## 2. Para cada uno de los 2.880 clientes que se encuentra en la base de datos final de CLIENTES, determinar cuáles tienen facturación registrada. Esto da 448 clientes

In [140]:
%cd D:\MINE\ProyectoFinalMINE\Datos\Fuentes Usadas Tableau\salidas

D:\MINE\ProyectoFinalMINE\Datos\Fuentes Usadas Tableau\salidas


In [141]:
cliente_con_sector = pd.read_excel('cliente_con_sector.xlsx')

In [142]:
facturacion_total.shape

(3731, 9)

## Los dataframe que haran merge son:
-client_fact_ultima_factura

-cliente_con_sector

In [146]:
cliente_con_sector["EMAIL"].nunique()

1509

In [149]:
cliente_con_sector.shape

(1509, 12)

In [148]:
cliente_con_sector=cliente_con_sector.drop_duplicates(subset="EMAIL",keep='last')

In [150]:
client_fact_ultima_factura.head(2)

EMAIL   SubTotal  Numero_facturas ultima_factura
0  ABOGADOANDRESLOPEZ@GMAIL.COM  1905653.4               14     2017-12-10
1         AC.COMPRAS1@GMAIL.COM   408000.0                1     2014-12-24

In [151]:
client_fact_ultima_factura.shape

(448, 4)

In [152]:
cliente_con_sector.shape

(1509, 12)

In [153]:
fuente_combinada.shape

(448, 15)

In [155]:
fuente_combinada["EMAIL"].nunique()

448

## 3. Para cada uno de estos 448 clientes recuperar los siguientes atributos de la base final de cliente

In [82]:
fuente_combinada =  pd.merge(client_fact_ultima_factura, cliente_con_sector, how='left', left_on=['EMAIL'],right_on=['EMAIL'])
fuente_combinada.head(2)

EMAIL   SubTotal  Numero_facturas ultima_factura  \
0  ABOGADOANDRESLOPEZ@GMAIL.COM  1905653.4               14     2017-12-10   
1         AC.COMPRAS1@GMAIL.COM   408000.0                1     2014-12-24   

     TipoUsuario Prioridad    producto         Tipo Cotizacion a enviar  \
0  Usuario_Final   3_Media  Sms_Salida  Marcacion_Automatica_CON_Precio   
1  Usuario_Final   3_Media  Sms_Salida            SMS_Correo_CON_Precio   

  comercial     Fecha creacion                 estado          Sector  \
0   mpineda 2015-06-30 08:35:14  en_proceso_Con_Venta       Politicos   
1    wrojas 2014-12-17 11:59:15               cerrado  SinIdentificar   

     ciudad   Periodo  Antiguedad  
0  Medellin  201506.0        34.0  
1    Bogota  201412.0        41.0

In [158]:
%cd D:\MINE\ProyectoFinalMINE\Datos\Fuentes Usadas Tableau\salidas
fuente_combinada.to_excel("fuente_combinada.xlsx")

D:\MINE\ProyectoFinalMINE\Datos\Fuentes Usadas Tableau\salidas


In [160]:
fuente_combinada.loc[(fuente_combinada['EMAIL'] =='TXCLOUDSAS@GMAIL.COM')]

EMAIL  SubTotal  Numero_facturas ultima_factura  \
440  TXCLOUDSAS@GMAIL.COM  972344.0                2     2016-06-07   

       TipoUsuario Prioridad    producto Tipo Cotizacion a enviar comercial  \
440  Usuario_Final    2_Alta  Sms_Salida    SMS_Correo_CON_Precio   fpineda   

        Fecha creacion                 estado         Sector ciudad   Periodo  \
440 2015-08-21 17:13:17  en_proceso_Con_Venta  Independiente  OTROS  201508.0   

     Antiguedad  
440        33.0

In [77]:
#%cd /home/xubuntu/Documents/ProyectoFinalMINE/Datos/Salidas
%cd D:\MINE\ProyectoFinalMINE\Datos\Fuentes Usadas Tableau\salidas
#fact_cliente_fecha.to_excel("fact_cliente_fecha.xlsx")

D:\MINE\ProyectoFinalMINE\Datos\Fuentes Usadas Tableau\salidas
